go/cmd/dolt/commands/send_metrics.go

```diff
@@ -2,47 +2,74 @@ package commands
 
 import (
 	"context"
+	"time"
 
 	"github.com/liquidata-inc/dolt/go/libraries/doltcore/dbfactory"
 	"github.com/liquidata-inc/dolt/go/libraries/doltcore/env"
 	"github.com/liquidata-inc/dolt/go/libraries/events"
-	"github.com/liquidata-inc/dolt/go/libraries/utils/filesys"
 )
 
 // SendMetricsCommand is the command used for sending metrics
-const SendMetricsCommand = "send-metrics"
-
-// var errMetricsDisabled = errors.New("metrics are currently disabled")
+const (
+	SendMetricsCommand = "send-metrics"
+	outputFlag         = "--output"
+)
 
 // SendMetrics is the commandFunc used that flushes the events to the grpc service
 func SendMetrics(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
 	disabled, err := events.AreMetricsDisabled(dEnv)
+	if err != nil {
+		// log.Print(err)
+		return 1
+	}
+
+	if disabled {
+		return 0
+	}
+
 	if !disabled && err == nil {
-		fs := filesys.LocalFS
+		ctx, cancel := context.WithTimeout(ctx, time.Minute)
+		defer cancel()
 
-		root, err := env.GetCurrentUserHomeDir()
+		root, err := dEnv.GetUserHomeDir()
 		if err != nil {
 			// log.Print(err)
 			return 1
 		}
 
 		dolt := dbfactory.DoltDir
 
-		eventGrpcFlush := events.NewEventGrpcFlush(fs, root, dolt, dEnv)
+		if len(args) > 0 {
+			if args[0] == outputFlag {
+				iof := events.NewIOFlusher(dEnv.FS, root, dolt, dEnv)
+
+				err := iof.Flush(ctx)
+				if err != nil {
+					if err == events.ErrFileLocked {
+						return 2
+					}
+
```

Comment:
rather than having 2 code paths that do similar things, its better to do something like

```
var flusher Flusher
if outputFlag {
    flusher = events.NewIOFlusher(dEnv.FS, root, dolt, dEnv)
}
else {
    flusher = events.NewGrpcEventFlusher(dEnv.FS, root, dolt, dEnv)
}

err = flusher.Flush(ctx)
... etc
```